packages/blade/src/components/List/List.tsx

```diff
@@ -102,18 +111,20 @@ const List = ({ variant = 'unordered', size, children, icon }: ListProps): React
 
   return (
     <ListProvider value={listContextValue}>
-      <ListElement
-        marginTop={level ? undefined : 'spacing.3'}
-        {...metaAttribute(MetaConstants.Component, MetaConstants.List)}
-        {...makeAccessible({ role: 'list' })} // Role needed for react-native
-      >
-        {variant === 'unordered'
-          ? childListItems
-          : childListItems.map(
-              (child, index) =>
-                React.cloneElement(child as React.ReactElement, { _itemNumber: index + 1 }), // adds _itemNumber for rendering ordered list bullets
-            )}
-      </ListElement>
+      <BaseBox {...getStyledProps(styledProps)}>
```

Comment:
Can't add it directly to ListElement?

Reply:
So the reason I added BaseBox is, it gets easier to maintain all the abstraction in BaseBox itself. We also avoid clashing any props or unknowingly exporting some styles that can change the UI of the component.

A lot of components are block components so adding BaseBox doesn't make different in terms of UI. There are some components like `<Text />`, `<Link />` where adding div was messing up with styles so for those cases in particular I have added styled-props directly to the component.

Reply:
In general libraries do `const Text = styled(BaseBox)()` and then pass `as="p"` as default on web. This is how all components end up getting the props. We can't exactly do that because `as` is not supported in React Native.

Reply:
Yup so strategy is to add Box wrapper wherever possible (mostly on block elements) or else add the props directly to component (for inline things like Text, Link, Svg.